internal/pkg/agent/vault/seed.go

```diff
@@ -0,0 +1,43 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package vault
+
+import (
+	"errors"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+)
+
+const seedFile = ".seed"
+
+func getSeed(path string) ([]byte, error) {
+	fp := filepath.Join(path, seedFile)
+	b, err := ioutil.ReadFile(fp)
```

Comment:
might be a good idea to check permissions of seed file and complain if they are too permissive.

Reply:
added the root owner check. the agent will fail to run in .seed file is not root/root owned
```
$ sudo ./elastic-agent -e
Error: could not read overwrites: could not initialize config store: non-root file owner
For help, please see our troubleshooting guide at https://www.elastic.co/guide/en/fleet/8.3/fleet-troubleshooting.html
```

Reply:
@nimarezainia, do we have plans to allow users to run the agent as non-root?

Reply:
> @nimarezainia, do we have plans to allow users to run the agent as non-root?

Not at the moment.